### ISSUE TITLE
CNTK APIs supporting .Net bindings?

### ISSUE THREAD

**Issue by openyang** [Microsoft/CNTK#817]
**FAQ page 2 days ago**
https://github.com/Microsoft/CNTK/wiki/CNTK-FAQ/234b2d517c90b5e9414904d1553cd5453ea9717b

September 2016 we plan to release the new CNTK APIs supporting both Python, and C++ bindings, so you will be able to configure your networks, create new nodes, as well as programmatically feed the data from your own Python, and C++ programs.

**FAQ page 13 days ago**
https://github.com/Microsoft/CNTK/wiki/CNTK-FAQ/289a2f6da3e5aba20ff400c7fe1b7c66b4e9ae29

September 2016 we plan to release the new CNTK APIs supporting both Python, C++, and .Net (C#) bindings, so you will be able to configure your networks, create new nodes, as well as programmatically feed the data from your own Python, C++, and .Net programs.

CNTK APIs supporting .Net bindings?

**Comment by epsilon-greedy** [comment]
This issue is causing me anxiety as I have a reinforcement learning project ready to start once CNTK V2 is released. The RL Environment element of my RL project is an existing business function comprised of +20,000 lines of C# hence a .NET CNTK RL Agent is the much preferred development route.

I was hoping to run my CNTK neural net function approximator, the Agent and the Environment in the same Windows .Net process and thus benefit from an Agent/Action/Reward loop of up to 100 iterations per second.

**Comment by frankseide** [comment]
We fell a little behind. The Python bindings are done in SWIG. I think that can be quickly repurposed into .Net bindings, once done. So it shouldn't be too far out.

**Comment by epsilon-greedy** [comment]
No worries about the delayed .Net bindings, it is reassuring to read the strategic thinking re. .Net CNTK APIs has not changed.

**Comment by StevenGann** [comment]
Hello from October! Is there still a plan for .NET?

**Comment by StevenGann** [comment]
Two days later. I'm sure there's more than just us 3 .NET developers interested in CNTK.

Just think what'll happen when Unity users get access to CNTK. Tensorflow may be better, but it's almost impossible to get running on Windows so it's useless for games and other mass-market applications. I'm picturing dozens of Unity and Monogame-based indie games all over Steam, boasting deep learning AI for NPCs.

.NET support could mean the biggest expansion of CNTK's userbase.

**Comment by andrewfry** [comment]
I am very interested in .net bindings. I was very excited when I read that
a beta would be available in September but then that was pushed back or
appears to be eliminated.

**Comment by JimSEOW** [comment]
Please check the latest .NET CNTK [links](https://github.com/Microsoft/CNTK/issues/960#issuecomment-306508050)

**Comment by deepakkumar1984** [comment]
Could be a useful project to use for developing model and training them: https://github.com/deepakkumar1984/SiaNet

A c# wrapper on CNTK backend for easy stacking of neural network.
